src/Microsoft.DotNet.Darc/src/DarcLib/Models/Darc/DependencyGraph.cs

```
@@ -0,0 +1,213 @@
+            ILogger logger, 
+            DependencyGraphNode node, 
+            IEnumerable<string> remotesMap, 
+            string reposFolder, 
+            string testPath = null)
+        {
+            try
+            {
+                IEnumerable<DependencyDetail> dependencies = null;
+
+                if (!string.IsNullOrEmpty(testPath))
+                {
+                    Local local = new Local(
+                        $"{testPath}/{node.DependencyDetail.RepoUri}/{node.DependencyDetail.Commit}/.git", 
+                        logger);
+                    dependencies = await local.GetDependenciesAsync();
+                }
+                else if (remote)
+                {
+                    Remote remoteClient = new Remote(darcSettings, logger);
+                    dependencies = await remoteClient.GetDependenciesAsync(
+                        node.DependencyDetail.RepoUri, 
+                        node.DependencyDetail.Commit);
+                }
+                else
+                {
+                    string repoPath = GetRepoPath(node.DependencyDetail, remotesMap, reposFolder, logger);
+
+                    // Local's constructor expects the repo's .git folder to be passed in. In this particular case we could 
+                    // pass any folder under 'repoPath' or even a fake one but we use .git to keep things consistent to what 
+                    // Local expects
+                    Local local = new Local($"{repoPath}/.git", logger);
+                    string fileContents = LocalHelpers.GitShow(
+                        repoPath, 
+                        node.DependencyDetail.Commit, 
+                        VersionFiles.VersionDetailsXml, 
+                        logger);
+                    dependencies = local.GetDependenciesFromFileContents(fileContents);
+                }
+
+                return dependencies;
+            }
+            catch
+            {
+                return null;
```

This seems like a bad idea... e.g. if the `git` command fails for a strange reason, we would just ignore the dependencies it would have gotten? Maybe even worse (more likely) for the Remote case?

---

We are not ignoring previous nodes.  Returning null is equivalent as no dependencies found which mean, in the stack traversing, the current node been visited has no dependencies but at the end, if everything else worked, a graph will be returned

---

But won't it be an incorrect graph if e.g. a network call failed?

---

I think I see what's your point. You are suggesting to better stop the execution if something failed instead of returning an incomplete graph. I think that makes sense

---

Yeah. If there's a strong use case for the incomplete graph, I think it'd be reasonable to output any errors to stderr, continue, and return a non-success exit code in the end, but silently ignoring all exceptions seems very bad.

---

When coding this I thought I'd better have an incomplete graph than no graph but after thinking a bit more about that it said graph would be used for something important then that could lead to weird decisions/behaviors,.. Hence the change